### PR TITLE
Convert mBuild custom group names

### PR DESCRIPTION
**Pull request by CalCraven**
Let the `from_mbuild` conversion allow users to specify custom ways to get the group names. This argument, which defaults to the standard GMSO group definitions, would allow users to arbitrarily section off parts of their compound and store that into the group attribute.

This label is generated by recursively moving top down through the hierarchy until a child is found with a matching label. Then, every particle in that child is given the specified site.group label. This is different than the ParmED residues definition, which moves up from a particle through `particle.parent`. It shouldn't cause an issue, but if you specify a label that is located in multiple places in your hierarchy, this would give slightly (more intuitive than ParmED imo) different behavior.

Errors are thrown if any site is missing a group label in this instance or if you supply group labels that are not used. 

- [x] unit tests
- [x] doc strings
- [x] functions for custom group labeling 
- [ ] issue raised

### CODE REVIEW

**Review by bc118 — APPROVED** [review]
This was also extensively tested outside of this PR.

**Review by daico007 — COMMENTED** [review]
Just one comment, the rest looks good to me

File: gmso/external/convert_mbuild.py [hunk 1]
```diff
@@ -63,39 +68,49 @@ def from_mbuild(
...
-        top.add_site(site)
+    def _assign_site_info(compound, top, site_map, search_method):
+        """Use site map to apply Compound info to Topology."""
+        for part in compound.particles():
+            site = _parse_site(site_map, part, search_method)
+            top.add_site(site)
+
+        return
+
+    _assign_site_info(compound, top, site_map, search_method)
```
[the remark this answers is not in the document]
Comment (CalCraven) [reply gmso/external/convert_mbuild.py:113]: Yes, that make sense. I was just using this to organize my code, so it ended up being a pretty simple function. Good catch!

**Review by daico007 — APPROVED** [review]

**Comment by CalCraven** [comment]
Ready to merge @daico007 ? I can close #688 with this as well.

**Comment by daico007** [comment]
it's ready

**Pull request merged**